db/flush_job.cc

```diff
@@ -370,6 +362,8 @@ Status FlushJob::MemPurge(autovector<MemTable*>& purged_mems) {
   // or at least range tombstones, copy over the info
   // to the new memtable.
   if (iter->Valid() || !range_del_agg->IsEmpty()) {
+    // Arbitrary heuristic: maxSize is 60% cpacity.
+    size_t maxSize = ((mutable_cf_options_.write_buffer_size + 6ULL) / 10ULL);
```

Comment:
`* 6U) / 10U`

Also, `42ULL` probably won't fit in size_t on 32-bit. Just be consistent about unsigned vs. signed in arithmetic, and promotion will generally take care of type sizes.

Reply:
Fixed!